README.md

```diff
@@ -17,6 +17,259 @@ Comming soon ;)
+
+The location of repository where you or your team mates push the project.
+```yml
+repository_url: https://github.com/<username>/<project-name>
+```
+
+##### `username`
+`string`
+
+The user used to access the servers through SSH.
+```yml
+user: USERNAME
+```
+_**NOTE:** This user will be used to access all the servers mentioned in stages below._
+
+##### `port`
+`integer`
+
+The port used to access the servers through SSH.
+```yml
+port: SERVER_SSH_PORT
+```
+_**NOTE:** This port will be used to access all the servers mentioned in stages below._
+
+### Stages
+Stages refer to the configured remote servers where you would like to deploy your application. E.g. `dev`, `qa`, `uat`, `staging`, `production`.
+
+##### `stages`
+`array/list`
+
+The list of different stages of your application with their details.
+```yml
+stages:
+    dev:
+        ...
+    uat:
+        ...
+```
+
+##### `stages`.`<stage>`.`host`
+`string`
+
+Address of the hosted server for the defined stage.
+```yml
+host: <stage>.your-app.com
+```
+
+##### `stages`.`<stage>`.`user` `optional`
+`string`
+
+The user to access the defined staged server. This user overrides the user defined in [basic configuration](https://github.com/kabirbaidhya/boss-cli#basic-configutation). If not defined, the user defined in [basic configuration](https://github.com/kabirbaidhya/boss-cli#basic-configutation) will be used.
+```yml
+user: <stage>_SERVER_USERNAME
+```
+
+##### `stages`.`<stage>`.`port` `optional`
+`integer`
+
+The port to access the defined staged server. This port overrides the port defined in [basic configuration](https://github.com/kabirbaidhya/boss-cli#basic-configutation). If not defined, the port defined in [basic configuration](https://github.com/kabirbaidhya/boss-cli#basic-configutation) will be used.
+```yml
+port: <stage>_SERVER_SSH_PORT
+```
+
+
+##### `stages`.`<stage>`.`public_url`
+`string`
+
+The public url to access this website.
+```yml
+public_url: http://dev.your-app.com
+```
+
+##### `stages`.`<stage>`.`app_dir`
+`string`
+
+The absolute path of the project root directory in the server.
+```yml
+app_dir: /path/to/your/app
+```yml
+
+##### `stages`.`<stage>`.`logging`
+`array/list`
+
+The list of files to view server logs.
+```yml
+logging:
+    ...
+```
+
+##### `stages`.`<stage>`.`logging`.`files`
+`array/list`
+
+The list of log files. It may include host server logs, database log and others.
+```yml
+files:
+    - /path/to/access/log/file
+    - /path/to/error/log/file
+    - /path/to/database/log/file
+```
+
+### Notifications
+You can configure to be notified when deployment starts to succeeds.
+
+##### `notifications`
+`array/list`
+
+The list of different chat/IRC clients through which notifications can be sent.
+```yml
+notifications:
+    slack:
+        ...
+    hipchat:
+        ...
+```
+
+Currently, only hipchat and slack are supported. Further integrations are welcome through pull-requests.
+
+#### Slack
+
+##### `notifications`.`slack`.`enabled`
```

Comment:
Let's prefer the format `slack.enabled` instead of `notifications`.`slack`.`enabled` for these kind of titles